packages/admin-ui/src/pages/dashboard/components/HostStats.tsx

```diff
@@ -2,41 +2,112 @@ import React, { useEffect } from "react";
 import { useApi } from "api";
 import Card from "components/Card";
 import ProgressBar from "react-bootstrap/ProgressBar";
+import humanFileSize from "utils/humanFileSize";
+import Loading from "../../../components/Loading";
+import { HostStatDisk } from "common";
 
 function parseVariant(value: number) {
   if (value > 90) return "danger";
   if (value > 75) return "warning";
   return "success";
 }
 
-function StatsCard({ id, percent }: { id: string; percent: string }) {
+/**
+ * Returns an element to be rendered containing diskUsed/diskTotal. If not returns 0/0
+ * @param  param0 Parameters of the diskUsed and diskTotal provided by statsDiskGet
+ */
+function DiskStats({ disk }: { disk: HostStatDisk }) {
+  return (
+    <p className="disk-usage">
+      {humanFileSize(parseInt(disk.used))}/
+      {humanFileSize(parseInt(disk.bBlocks))}
+    </p>
+  );
+}
+
+function StatsCard({
+  title,
+  percent,
+  disk
+}: {
+  title: string;
+  percent: string;
+  disk?: HostStatDisk;
+}) {
   const value = parseInt(percent);
+
   return (
     <Card className="stats-card">
       <div className="header">
-        <span className="id">{id}</span> <span className="usage">usage</span>
+        <span className="id">{title}</span> <span className="usage">usage</span>
       </div>
-      <ProgressBar variant={parseVariant(value)} now={value} label={percent} />
+      <ProgressBar
+        variant={parseVariant(value)}
+        now={value}
+        label={percent + "%"}
+      />
+      {disk ? <DiskStats disk={disk} /> : null}
     </Card>
   );
 }
-
 export function HostStats() {
-  const stats = useApi.getStats();
+  const cpuStats = useApi.statsCpuGet();
+  const memoryStats = useApi.statsMemoryGet();
+  const diskStats = useApi.statsDiskGet();
+
+  useEffect(() => {
+    const interval = setInterval(cpuStats.revalidate, 4 * 1000);
+    return () => {
+      clearInterval(interval);
+    };
+  }, [cpuStats]);
+
+  useEffect(() => {
+    const interval = setInterval(diskStats.revalidate, 1000);
+    return () => {
+      clearInterval(interval);
+    };
+  }, [diskStats]);
 
   useEffect(() => {
-    const interval = setInterval(stats.revalidate, 5 * 1000);
+    const interval = setInterval(memoryStats.revalidate, 1000);
     return () => {
       clearInterval(interval);
     };
-  }, [stats]);
+  }, [memoryStats]);
 
   return (
     <div className="dashboard-cards">
-      {stats.data &&
-        Object.entries(stats.data).map(([id, percent]) => (
-          <StatsCard key={id} id={id} percent={percent} />
-        ))}
+      {cpuStats.data ? (
+        <StatsCard key={0} title={"cpu"} percent={cpuStats.data.used} />
+      ) : cpuStats.error ? (
+        <Loading steps={["Not possible to load cpu"]} />
```

Comment:
In case of error you want to show an error component instead of a loading indicator. You can use the `ErrorView` component for that